sentry-android-core/src/main/java/io/sentry/android/core/internal/util/PrivilegeEscalationViaContentProviderChecker.java

```diff
@@ -0,0 +1,63 @@
+package io.sentry.android.core.internal.util;
+
+import android.annotation.TargetApi;
+import android.content.ContentProvider;
+import android.net.Uri;
+import android.os.Build;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+import io.sentry.NoOpLogger;
+import io.sentry.android.core.BuildInfoProvider;
+
+/**
+ * Protects against "Privilege Escalation via Content Provider" (CVE-2018-9492).
+ * <p>
+ * See https://www.cvedetails.com/cve/CVE-2018-9492/
+ * and https://github.com/getsentry/sentry-java/issues/2460
+ */
+@ApiStatus.Internal
+public final class PrivilegeEscalationViaContentProviderChecker {
```

Comment:
It could make sense to give this utility class a more generic name, so we could extends it's functionality in case there will be future CVEs.
```suggestion
public final class ContentProviderSecurityChecker {
```

Reply:
I would typically no here and drop the ol' [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it). I don't see future CVEs popping up specifically for ContentProviders when we've been scanning our apps with Secure Theorem for the past 2+ years and this is the only complaint we got so far.

However, I do like the generic name you suggested more in terms of length and how less intimidating it is 😃 

Sure! I'll rename to `ContentProviderSecurityChecker` !

Reply:
Done! https://github.com/getsentry/sentry-java/pull/2466/commits/17521025dd027c24721d518bcbbd3dab7f6bdaf6